### PR TITLE
Expand incremental pipeline test coverage

### PR DESCRIPTION
**Pull request by ahojnnes**
## Summary
- Add tests for `IncrementalPipelineOptions` propagation methods (`Mapper()`, `Triangulation()`, `LocalBundleAdjustment()`, `GlobalBundleAdjustment()`)
- Add tests for `CheckRunGlobalRefinement` conditions (frames ratio/freq, points ratio/freq)
- Add tests for snapshot writing during reconstruction, max runtime stopping, stop callback interruption
- Add tests for `TriangulateReconstruction`, `DatabaseCache` constructor path, and null argument validation
- Targets item #8 from test coverage opportunities (`controllers/incremental_pipeline.cc`, 78.2% coverage, 84 uncovered lines)

## Test plan
- [ ] CI passes all existing and new tests
- [ ] New tests build and run successfully

### CODE REVIEW

File: src/colmap/controllers/incremental_pipeline_test.cc [hunk 1]
```diff
@@ -669,5 +671,427 @@ TEST(IncrementalPipeline, PriorBasedSfMWithRandomSeedStability) {
...
+  EXPECT_FALSE(pipeline.CheckRunGlobalRefinement(reconstruction, 4, 50));
+
+  // Trigger via frames ratio only: 4 >= 100.0 * 0 = 0 -> true.
+  // Use prev=0 so ratio*0=0 makes it trigger.
+  EXPECT_TRUE(pipeline.CheckRunGlobalRefinement(reconstruction, 0, 50));
+
+  // Trigger via frames freq: set freq=2, prev=2 -> 4 >= 2 + 2 = 4 -> true.
+  options->ba_global_frames_freq = 2;
+  EXPECT_TRUE(pipeline.CheckRunGlobalRefinement(reconstruction, 2, 50));
+  options->ba_global_frames_freq = 100000;  // Reset.
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/controllers/incremental_pipeline_test.cc:841]: Keeping tests self-contained for clarity. The repetition makes each test case independently readable.

File: src/colmap/controllers/incremental_pipeline_test.cc [hunk 2]
```diff
@@ -669,5 +671,427 @@ TEST(IncrementalPipeline, PriorBasedSfMWithRandomSeedStability) {
...
+  auto options = std::make_shared<IncrementalPipelineOptions>();
+  // Set an extremely short runtime to trigger the timeout path.
+  // The pipeline should stop and keep whatever reconstruction it has.
+  // max_runtime_seconds must be > 0 for the check to trigger (the check is
+  // max_runtime_seconds > 0 && elapsed > max_runtime_seconds).
+  options->max_runtime_seconds = 1;
+  IncrementalPipeline pipeline(options, database, reconstruction_manager);
+  pipeline.Run();
+
+  // With max_runtime_seconds=0, the pipeline should stop almost immediately.
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/controllers/incremental_pipeline_test.cc:961]: Keeping tests self-contained for clarity. The repetition makes each test case independently readable.

File: src/colmap/controllers/incremental_pipeline_test.cc [hunk 3]
```diff
@@ -669,5 +671,427 @@ TEST(IncrementalPipeline, PriorBasedSfMWithRandomSeedStability) {
...
+
+  auto reconstruction_manager = std::make_shared<ReconstructionManager>();
+  auto options = std::make_shared<IncrementalPipelineOptions>();
+  IncrementalPipeline pipeline(options, database, reconstruction_manager);
+
+  // Set the stop function to immediately stop after the initial pair.
+  int callback_count = 0;
+  pipeline.SetCheckIfStoppedFunc([&callback_count]() {
+    // Stop after the first check (i.e., right after initialization attempt).
+    return ++callback_count > 1;
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/controllers/incremental_pipeline_test.cc:1028]: Keeping tests self-contained for clarity. The repetition makes each test case independently readable.

File: src/colmap/controllers/incremental_pipeline_test.cc [hunk 4]
```diff
@@ -669,5 +671,427 @@ TEST(IncrementalPipeline, PriorBasedSfMWithRandomSeedStability) {
...
+
+TEST(IncrementalPipeline, NullDatabaseThrows) {
+  auto options = std::make_shared<IncrementalPipelineOptions>();
+  auto reconstruction_manager = std::make_shared<ReconstructionManager>();
+  std::shared_ptr<Database> null_database;
+
+  EXPECT_THROW(
+      IncrementalPipeline(options, null_database, reconstruction_manager),
+      std::invalid_argument);
+}
```
[the remark this answers is not in the document]
Comment (ahojnnes) [reply src/colmap/controllers/incremental_pipeline_test.cc:1092]: Keeping tests self-contained for clarity. The repetition makes each test case independently readable.
